aws-codeartifact-domain/src/main/java/software/amazon/codeartifact/domain/BaseHandlerStd.java

```diff
@@ -47,40 +45,54 @@
-          }
-          return awsResponse;
-        })
-        .stabilize((awsRequest, awsResponse, client, model, context) -> domainPolicyIsStabilized(model, client, request))
-        .progress();
+
+      final ResourceModel desiredModel = progress.getResourceModel();
+      final ResourceModel previousModel = request.getPreviousResourceState();
+
+      if (desiredModel.getPermissionsPolicyDocument() == null || policyIsUnchanged(desiredModel, previousModel)) {
+          return ProgressEvent.progress(desiredModel, callbackContext);
+      }
+      return proxy.initiate("AWS-CodeArtifact-Domain::Update::PutDomainPermissionsPolicy", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+          .translateToServiceRequest(Translator::translatePutDomainPolicyRequest)
+          .makeServiceCall((awsRequest, client) -> {
+              PutDomainPermissionsPolicyResponse awsResponse = null;
+              try {
+                  awsResponse = client.injectCredentialsAndInvokeV2(awsRequest, client.client()::putDomainPermissionsPolicy);
+                  logger.log("Domain permission policy successfully added.");
+              } catch (final AwsServiceException e) {
+                  String domainName = desiredModel.getDomainName();
+                  Translator.throwCfnException(e, Constants.PUT_DOMAIN_POLICY, domainName);
+              }
+              return awsResponse;
+          })
+          .progress();
   }
 
-    private boolean domainPolicyIsStabilized(
+    boolean policyIsUnchanged(final ResourceModel desiredModel, final ResourceModel previousModel) {
+        if (previousModel == null) {
+            return false;
+        }
+
+        if (desiredModel.getPermissionsPolicyDocument() == null || previousModel.getPermissionsPolicyDocument() == null) {
+            return false;
+        }
+        JsonNode desiredPolicy = MAPPER.valueToTree(desiredModel.getPermissionsPolicyDocument());
+        JsonNode currentPolicy = MAPPER.valueToTree(previousModel.getPermissionsPolicyDocument());
+        return desiredPolicy.equals(currentPolicy);
+    }
+
+    protected boolean doesDomainExist(
         final ResourceModel model,
         final ProxyClient<CodeartifactClient> proxyClient,
         ResourceHandlerRequest<ResourceModel> request
     ) {
-      try {
-          proxyClient.injectCredentialsAndInvokeV2(
-              Translator.translateToGetDomainPermissionPolicy(model, request), proxyClient.client()::getDomainPermissionsPolicy);
-          return true;
-      } catch (ResourceNotFoundException e) {
-          return false;
-      }
+        try {
+            proxyClient.injectCredentialsAndInvokeV2(
+                Translator.translateToReadRequest(model, request), proxyClient.client()::describeDomain);
+            return true;
+        } catch (ResourceNotFoundException e) {
+            return false;
+        }
     }
```

Comment:
not sure if this little green squares mean you have weird whitespace or something. Maybe accidentally had a couple of tabs instead of spaces?